cin_validator/rules/cin2022_23/rule_8530Q.py

```diff
@@ -0,0 +1,167 @@
+"""
+Rule number: 8530Q
+Module: Child Identifiers
+Rule details: If present <ExpectedPersonBirthDate> (N00098) should be between [<ReferenceDate> (N00603) minus 30 days] and [<ReferenceDate> (N00603) plus 9 months]
+Rule message: Please check: Expected Date of Birth is outside the expected range for this census (March to December of the Census Year end)
+"""
+from typing import Mapping
+
+import pandas as pd
+import datetime as dt
+
+from cin_validator.rule_engine import (
+    CINTable,
+    IssueLocator,
+    RuleContext,
+    RuleType,
+    rule_definition,
+)
+from cin_validator.test_engine import run_rule
+from cin_validator.utils import make_census_period
+
+# Get tables and columns of interest from the CINTable object defined in rule_engine/__api.py
+
+ChildIdentifiers = CINTable.ChildIdentifiers
+ExpectedPersonBirthDate = ChildIdentifiers.ExpectedPersonBirthDate
+
+Header = CINTable.Header
+ReferenceDate = Header.ReferenceDate
+
+# define characteristics of rule
+@rule_definition(
+    code="8530Q",
+    module=CINTable.ChildIdentifiers,
+    rule_type=RuleType.QUERY,
+    message="Please check: Expected Date of Birth is outside the expected range for this census (March to December of the Census Year end)",
+    affected_fields=[ExpectedPersonBirthDate],
+)
+def validate(
+    data_container: Mapping[CINTable, pd.DataFrame], rule_context: RuleContext
+):
+    df = data_container[ChildIdentifiers]
+    df.index.name = "ROW_ID"
+
+    df_ref = data_container[Header]
+    ref_date_series = df_ref[ReferenceDate]
+    collection_start, collection_end = make_census_period(ref_date_series)
+
+    #  <ExpectedPersonBirthDate> (N00098) should be between [<ReferenceDate> (N00603) minus 30 days] and [<ReferenceDate> (N00603) plus 9 months]
+
+    # Filter to only those with expected birthdate
+    df = df[~df[ExpectedPersonBirthDate].isna()]
+
+    # Find the reference date - 30
+    earliest_date = collection_start - dt.timedelta(30)
+    # Find reference date + 9 months
+    latest_date = collection_end + dt.timedelta(9 * 30)
+
```

Comment:
If the 9 months needs to be calculated more accurately we will need an extra package installing ( `from dateutil.relativedelta import relativedelta` )

Reply:
@phoebefryer You can use   collection_end + pd.DateOffset(months=9)
It also has days, weeks, years etc....

And it's a part of the pandas module so there's no need to install anything extra.

Reply:
Of course pandas has something to do the job! I've amended it now to use that instead

Reply:
The rule looks like it's looking at ReferenceDate, not collection start and end. I think the idea is that if the expected birth date isn't a month before or 9 months after the reference date, either you'd have a birth date, or you wouldn't know you were pregnant to have an expected birth date?